core/state/statedb.go

```diff
@@ -337,11 +337,14 @@ func (s *StateDB) GetProofByHash(addrHash common.Hash) ([][]byte, error) {
 // GetStorageProof returns the Merkle proof for given storage slot.
 func (s *StateDB) GetStorageProof(a common.Address, key common.Hash) ([][]byte, error) {
 	var proof proofList
-	trie := s.StorageTrie(a)
+	trie, err := s.StorageTrie(a)
+	if err != nil {
+		return nil, err
+	}
 	if trie == nil {
-		return proof, errors.New("storage trie for requested address does not exist")
+		return nil, errors.New("storage trie for requested address does not exist")
 	}
-	err := trie.Prove(crypto.Keccak256(key.Bytes()), 0, &proof)
+	err = trie.Prove(crypto.Keccak256(key.Bytes()), 0, &proof)
```

Comment:
Perhaps just to keep consistent error handling, maybe do

```if err != nil {
  return nil, err
}
return proof, nil
```

?

Thinking that if above you swapped out the uninitialized `proof` to `nil`, we might want to do it here too to be on the same style.